File: Notification-Service/src/main/java/byrd/riley/notificationservice/model/TwilioTexter.java
package byrd.riley.notificationservice.model;

import com.twilio.http.TwilioRestClient;
import com.twilio.rest.api.v2010.account.Message;
import com.twilio.type.PhoneNumber;

public class TwilioTexter implements Texter {

	private final TwilioRestClient client;
	private final SMSConfigurable config;
	
	public TwilioTexter() { this(new SMSConfiguration()); }

	public TwilioTexter(SMSConfigurable config) {
		this.config = config;
		client = (new TwilioRestClient.Builder(
			this.config.getAccountSID(),
			this.config.getAuthToken()
		)).build();
	}
	
	@Override
	public void sendText(String receiver, String messageBody) {
		sendText(receiver, config.getDefaultSenderPhone(), messageBody);
	}

	@Override
	public void sendText(String receiver, String sender, String messageBody) {
		Message message = Message.creator(
			new PhoneNumber(receiver),
			new PhoneNumber(sender),
			messageBody
		).create(client);
		
		System.out.println("Error " + message.getErrorCode() + ": " + message.getErrorMessage());
	}
}

File: Notification-Service-Kotlin/src/main/kotlin/byrd/riley/notificationservicekotlin/model/TwilioTexter.kt
package byrd.riley.notificationservicekotlin.model

import com.twilio.http.TwilioRestClient
import com.twilio.rest.api.v2010.account.Message
import com.twilio.type.PhoneNumber

class TwilioTexter(private val config: SMSConfigurable) : Texter {
    private val client = TwilioRestClient.Builder(
        config.accountSID,
        config.authToken
    ).build()

    constructor(): this(SMSConfiguration())

    override fun sendText(receiver: String, messageBody: String) {
        sendText(receiver, config.defaultSenderPhone, messageBody)
    }

    override fun sendText(receiver: String, sender: String, messageBody: String) {
        val message = Message.creator(
            PhoneNumber(receiver),
            PhoneNumber(sender),
            messageBody
        ).create(client)

        print("Error " + message.errorCode + ": " + message.errorMessage)
    }

}